Numbers game

aiidanm opened the pull request:
Not updated the main branch since very early on.

aiidanm committed:
Updated Styling

Updated div styling surrounding number

aiidanm committed:
Worked on collecting user input

Worked on collecting user input to then compare

aiidanm committed:
Working on checking guess to target number

Trying to check if entered number matches the generated but if statement doesnt seem to be working so far.

aiidanm committed:
Will now check if you enter the correct number

Can now type in your guess and it will console log if correct or not

aiidanm committed:
number now expands on each click

Added 1 digit to number on each press of new number, combined guessing and next number in one button and made code generally more compact

ryanbarouki commented on src/components/Games/Number/NumberMain.js:
```diff
@@ -47,28 +60,43 @@ const Input = styled.input`
 
 export const NumberMain = () => {
-  const [num, setNum] = useState(0);
+  const [num, setNum] = useState(Math.floor(Math.random() *  (9 - 1 +1)) +1);
```
you've got 9 - 1 + 1 you can remove the -1 + 1

ryanbarouki replied:
Actually probs want to replace with this 
`const [num, setNum] = useState(RandomNumberInRange(0,9));` 
as you want the number 0 included too

ryanbarouki commented on src/components/Games/Number/NumberMain.js:
```diff
@@ -47,28 +60,43 @@ const Input = styled.input`
+  const [num, setNum] = useState(Math.floor(Math.random() *  (9 - 1 +1)) +1);
+  const [newNum, setNewNum] = useState(Math.floor(Math.random() *  (9 - 1 +1)) +1);
 
   function RandomNumberInRange(min, max) {
```
Move this function outside of the component as it's more of a utility and doesn't require access to the state.

ryanbarouki commented on src/components/Games/Number/NumberMain.js:
```diff
@@ -47,28 +60,43 @@ const Input = styled.input`
+
+  };
+
+ const [guess, setGuess] = useState('');
```
Usually put all state stuff together at the top

ryanbarouki commented on src/components/Games/Number/NumberMain.js:
```diff
@@ -47,28 +60,43 @@ const Input = styled.input`
 export const NumberMain = () => {
-  const [num, setNum] = useState(0);
+  const [num, setNum] = useState(Math.floor(Math.random() *  (9 - 1 +1)) +1);
+  const [newNum, setNewNum] = useState(Math.floor(Math.random() *  (9 - 1 +1)) +1);
```
You don't really need this as state, you can generate a new random number on click and add it to the `num` state

ryanbarouki commented on src/components/Games/Number/NumberMain.js:
```diff
@@ -47,28 +60,43 @@ const Input = styled.input`
-  const handleClick = () => {
-    setNum(RandomNumberInRange(1, 999999999));
+  const handleClick = (e) => {
+    setNewNum(RandomNumberInRange(1, 9));
+    setNum(String(num) + String(newNum)); 
```
`setNum(num + RandomNumberInRange(0,9))` If you make `RandomNumberInRange` return a string instead of int then you won't have to keep converting. You can just keep everything as a string because you only ever need it as that.

aiidanm committed:
Update Number game

Simplified code after comment review.

removed redundant const, moved function out of component
